localstack/services/stepfunctions/stepfunctions_starter.py

```diff
@@ -107,9 +108,20 @@ def check_stepfunctions(expect_shutdown=False, print_error=False):
         assert out and isinstance(out.get("stateMachines"), list)
 
 
-def restart_stepfunctions():
-    if not PROCESS_THREAD:
+def restart_stepfunctions(persistence_path: Optional[str] = None):
+    if not PROCESS_THREAD or not PROCESS_THREAD.process:
         return
     LOG.debug("Restarting StepFunctions process ...")
+    pid = PROCESS_THREAD.process.pid
     PROCESS_THREAD.stop()
-    start_stepfunctions(asynchronous=True)
+    _wait_for_process_to_be_killed(pid)
```

Comment:
Alternatively we could also wait for the port to be closed, as that is usually a good indicator that the process is terminated (and for restarting we primarily need the network port to be free again).